kornia/augmentation/augmentation.py

```diff
@@ -240,3 +249,37 @@ def forward(self, images: torch.Tensor) -> torch.Tensor:  # type: ignore
             self.erase_scale_range,
             self.aspect_ratio_range
         )
+
+
+class RandomPerspective(AugmentationBase):
+    r"""Performs Perspective transformation of the given torch.Tensor randomly with a given probability.
+
+    Args:
+        p (float): probability of the image being perspectively transformed. Default value is 0.5
+        distortion_scale(float): it controls the degree of distortion and ranges from 0 to 1. Default value is 0.5.
+        return_transform (bool): if ``True`` return the matrix describing the transformation
+        applied to each. Default: False.
+        input tensor.
+    """
+
+    def __init__(self, distortion_scale: float = 0.5, p: float = 0.5, return_transform: bool = False) -> None:
+        super(RandomPerspective, self).__init__(F.apply_perspective, return_transform)
+        self.p: float = p
+        self.distortion_scale: float = distortion_scale
+        self.return_transform: bool = return_transform
+
+    def __repr__(self) -> str:
+        repr = f"(distortion_scale={self.distortion_scale}, p={self.p}, return_transform={self.return_transform})"
+        return self.__class__.__name__ + repr
+
+    @staticmethod
+    def get_params(batch_size: int, height: int, width: int, p: float,
+                   distortion_scale: float) -> Dict[str, torch.Tensor]:
+        return pg._random_perspective_gen(batch_size, height, width, p, distortion_scale)
+
+    def forward(self, input: UnionType, params: Optional[Dict[str, torch.Tensor]] = None) -> UnionType:  # type: ignore
```

Comment:
Is the Optinal keyword really necessary?

Reply:
it's recommended by mypy gods when type is none